MV.Wildebeest.Core/source/core/java/co/mv/wb/impl/WildebeestApiImpl.java

```diff
@@ -407,6 +406,15 @@ public void migrate(
 
 		List<Migration> path = paths.get(0);
 
+		try
+		{
+			validateMigrationStates(resource);
+		}
+		catch (MigrationInvalidStateException e)
+		{
+			throw new RuntimeException(e);
```

Comment:
Validation exceptions like the new MigrationInvalidStateException and existing ones like InvalidStateSpecifiedException are checked exceptions because we want them to bubble up to the interface (CLI / API) to be presented back to the caller.  You can see an example of how this is handled for the CLI in the run() method in WildebeestCommand.

Most likely rather than catching and wrapping this as an unchecked exception, you want to declare it in the throws and bubble it up to be presented through the CLI in the same way.